src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -94,7 +94,16 @@ public Task<Session> TryCreateServiceSessionAsync(string serviceName, Solution s
         public async Task<Session> TryCreateServiceSessionAsync(string serviceName, Solution solution, object callbackTarget, CancellationToken cancellationToken)
         {
             var snapshot = await GetPinnedScopeAsync(solution, cancellationToken).ConfigureAwait(false);
```

Comment:
:memo: Moving the method caused this line to appear in two locations while removing the need for two instances of calls to `Dispose`. Looking at the diff for both approaches, this one seems cleaner.